sqlglot/dialects/hive.py

```diff
@@ -335,13 +338,17 @@ class Generator(generator.Generator):
             exp.TableFormatProperty: exp.Properties.Location.POST_SCHEMA,
         }
 
-        def with_properties(self, properties):
+        def arrayagg_sql(self, expression: exp.ArrayAgg) -> str:
+            arg = expression.this if isinstance(expression.this, exp.Order) else expression
+            return f"COLLECT_LIST({self.sql(arg, 'this')})"
```

Comment:
self.func("COLLECT_LIST", expression.this.this if isinstance(expression.this, exp.Order) else expression.this)